sfaira/data/dataloaders/base/annotation_container.py

```diff
@@ -102,8 +102,12 @@ def read_from_yaml(self, yaml_path: Union[str, None], sample_fn: Union[None, str
             for k, v in yaml_vals["attr"].items():
                 if v is not None and k not in ["organism", "sample_fns"]:
                     if isinstance(v, dict):  # v is a dictionary over file-wise meta-data items
-                        assert sample_fn in v.keys(), f"did not find key {sample_fn} in yamls keys for {k}"
-                        v = v[sample_fn]
+                        if sample_fn in v.keys():
+                            # only set value if field exists
+                            v = v[sample_fn]
+                        else:
+                            # otherwise skip field
+                            continue
```

Comment:
Does this continue statement continue the outer for loop? If so, the following line is unreachable right?

Reply:
It should continue this for loop: `for k, v in yaml_vals["attr"].items()`. But actually you don't really need the `continue`, right? `if sample_fn in v.keys()` should be enough

Reply:
I would just remove the whole `else` block

Reply:
Yes agree it seems unnecessary